research-software-directory/RSD-as-a-service
# Improvements to organisation pages

# Improvements to organisation pages

Closes #720 

Changes proposed in this pull request:
* Search box on software and project page of an organisation is moved from page header to content of the page (see screenshot 1)
* Software releases are loaded per year.  Selection of the year is done by clicking on the year button at the top of the page. Selected year is saved to url query and can be deep linked (see screenshot 2). Software releases of parent include releases of "children" organisations (research units)
* Breadcrumb navigation labels are set to max width of 20rem. Long labels use ellipis (...). Complete name is shown on mouseover. The breadcrumb segment will be collapsed when more than 3 segments (dept) are used. (see screenshot 3)
* Hover style is removed from unit list item to avoid confusion about item being "clickable". Each list item can have up to  3 actions (rsd url, website url and edit button), therefore it might not be clear to user which action should be the "default" action. At the moment I choose to remove hover element to avoid confusion.

How to test:
* `make start` to rebuild everything
* `docker-compose down && docker-compose up` to restart app with scrapers
*  give some time (10 min.) to scraper to scrape releases (or run `docker-compose exec scrapers java -cp /usr/myjava/scrapers.jar nl.esciencecenter.rsd.scraper.doi.MainReleases` a few times). In the meantime you can validate other changes
* navigate to organisation software page. Confirm that search on software page is present and finds only software within the organisation
* navigate to organisation project page. Confirm search on project page works
* login as rsd_admin (professor3 if you have same env settings as me).
* navigate to organisation units and add new unit
* navigate to new unit you just created, confirm no hover styles are used on the whole list element. You will need to click on the unit name to visit the unit.
* add the child unit to this new unit (third level) with long name
* navigate to child unit. confirm breadcrumb labels length and collapsing when more than 3 segments are used
* add software to second level child
* add software to first level child
* validate that parent includes children software and releases
* validate that citation section is shown on software page with all releases listed

## Search moved to software and project page body

![image](https://user-images.githubusercontent.com/9204081/218567319-743c7b79-c2bf-4899-9733-9c133db357e1.png)

## Software releases loaded per year
![image](https://user-images.githubusercontent.com/9204081/218066937-18742eea-83c3-4c2c-b441-011059ce75c5.png)

## Breadcrumb label with max. width 20rem and collapsed when > 3 segments
![image](https://user-images.githubusercontent.com/9204081/218067427-97af9cae-94c6-45aa-a90c-c2044bbaf142.png)

![image](https://user-images.githubusercontent.com/9204081/218067499-1325b7de-8d99-442c-90ee-aaaca5c85819.png)


PR Checklist:
